### PR TITLE
Add rule for Service Need Indicator - 186711621

### PR DESCRIPTION
**Pull request by dtgreiner**
[//]: # 'remove this if PR is for a release-* branch'
# _Please squash merge this PR_

## Description

Add New Rul for Service Need Indicators

## Type of change
[//]: # 'remove options that are not relevant'
- [ ] Bug fix
- [X] New feature (adds functionality)
- [ ] Code clean-up / housekeeping
- [ ] Dependency update

## Checklist before requesting review
- [X] I have performed a self-review of my code
- [X] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [X] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [X] My code follows the style guidelines of this project (rubocop)
- [X] I have updated the documentation (or not applicable)
- [X] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: app/models/rules/service_need.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Rules::ServiceNeed < Rule
+  def clients_that_fit(scope, requirement, opportunity)
+    if Client.column_names.include?(:service_need.to_s)
+      scope.where(service_need: requirement.positive)
+    else
+      raise RuleDatabaseStructureMissing.new("clients.enrolled_in_th missing. Cannot check clients against #{self.class}.")
```
Comment (eanders) [inline app/models/rules/service_need.rb]: Ah, this probably should say:
```suggestion
      raise RuleDatabaseStructureMissing.new("clients.service_need missing. Cannot check clients against #{self.class}.")
```

Comment (eanders) [reply app/models/rules/service_need.rb]: I think I missed that on the last rule

**Review by eanders — APPROVED** [review]
Looks good pending that one small change.

File: app/models/rules/service_need.rb (hunk 1 above)
Comment (dtgreiner) [reply app/models/rules/service_need.rb]: I've got this updated in the current PR and will create a new change for the housing barrier rule

Comment (eanders) [reply app/models/rules/service_need.rb]: Thanks!

**Pull request merged**